src/Rector/Class_/RemoveEmptyTableAttributeRector.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Doctrine\Rector\Class_;
+
+use Doctrine\ORM\Mapping\Entity;
+use Doctrine\ORM\Mapping\Table;
+use PhpParser\Node;
+use PhpParser\Node\Stmt\Class_;
+use Rector\Core\Rector\AbstractRector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Rector\Doctrine\Tests\Rector\Class_\RemoveEmptyTableAttributeRector\RemoveEmptyTableAttributeRectorTest
+ */
+final class RemoveEmptyTableAttributeRector extends AbstractRector
+{
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Remove empty Table attribute on entities because it\'s useless',
+            [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+    use Doctrine\ORM\Mapping as ORM;
+
+    #[ORM\Table]
+    #[ORM\Entity]
+    class Product
+    {
+    }
+    CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+    use Doctrine\ORM\Mapping as ORM;
+
+    #[ORM\Entity]
+    class Product
+    {
+    }
+    CODE_SAMPLE
+                ),
+            ]
+        );
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [Class_::class];
+    }
+
+    /**
+     * @param Class_ $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        $removed = false;
+
+        if (!$this->isEntity($node)) {
+            return null;
+        }
+
+        foreach ($node->attrGroups as $attrGroup) {
+            foreach ($attrGroup->attrs as $attribute) {
+                if (!$this->nodeNameResolver->isName($attribute, Table::class)) {
```

Comment:
use string name instead of `::class` to avoid converted to prefixed value on downgrade:

```suggestion
                if (!$this->nodeNameResolver->isName($attribute, 'Doctrine\ORM\Mapping\Table')) {
```